### DISCUSSION TITLE
Triton test failed when running models with Pytorch backend

### DISCUSSION THREAD

**Discussion by cmpute** [InternLM/lmdeploy#2113] [Q&A] [answered]
![image](https://github.com/user-attachments/assets/2b0ada9c-0cf7-4a8e-9101-7c78a92dc4e9)
I installed lmdeploy from pip, and there is no cuda compiler in the system. Does triton requires a cuda compiler?

**Comment by grimoire** [comment]
The answer to "Does triton require a Cuda **compiler**?" depends on your NVIDIA driver version. Triton has a prepackaged PTXAS, it can be used if you have a compatible driver. Or you need to install cuda manually and set `TRITON_PTXAS_PATH`.

Your log indicates that you don't have a compatible Cuda **runtime**. Triton needs to link the runtime library to build a kernel launcher.

**Reply by cmpute** [reply]
Thanks for the reply! Do I need a system CUDA installation? I have installed a nvcc compiler in a conda environment, but still failed to use triton

**Reply by grimoire** [reply]
The conda environment should be ok. You can add the cuda runtime (libcuda.so) folder path to LIBRARY_PATH.

**Comment by cmpute** [comment, marked as answer]
This is a issue for triton, I commented on the official repo here: https://github.com/triton-lang/triton/issues/1617#issuecomment-2246858493
